data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineParser.java

```diff
@@ -48,10 +49,14 @@ public class PipelineParser {
     private final String pipelineConfigurationFileLocation;
     private final Map<String, PipelineConnector> sourceConnectorMap = new HashMap<>(); //TODO Remove this and rely only on pipelineMap
     private final PluginFactory pluginFactory;
+    private final PeerForwarder peerForwarder;
 
-    public PipelineParser(final String pipelineConfigurationFileLocation, final PluginFactory pluginFactory) {
+    public PipelineParser(final String pipelineConfigurationFileLocation,
+                          final PluginFactory pluginFactory,
+                          final PeerForwarder peerForwarder) {
         this.pipelineConfigurationFileLocation = pipelineConfigurationFileLocation;
         this.pluginFactory = Objects.requireNonNull(pluginFactory);
+        this.peerForwarder = peerForwarder;
```

Comment:
Are we not doing a null check on `peerForwarder`?

Reply:
I'm not sure if this check is required, because during bean creation we don't return null. But I'll add it to have a check here.